backend/plan/views.py

```diff
@@ -176,6 +183,106 @@ def recommend_courses_view(request):
         status=status.HTTP_200_OK,
     )
 
+class PrimaryScheduleViewSet(viewsets.ViewSet):
+    model = PrimarySchedule
+    queryset = PrimarySchedule.objects.none()
+    http_method_names = ["get", "post"]
```

Comment:
Can we actually change post to put (idempotent) - make sure to change all POST references below if so